src/shared_modules/content_manager/src/components/zipDecompressor.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Wazuh content manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * November 03, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _ZIP_DECOMPRESSOR_HPP
+#define _ZIP_DECOMPRESSOR_HPP
+
+#include "json.hpp"
+#include "updaterContext.hpp"
+#include "utils/chainOfResponsability.hpp"
+#include "utils/zlibHelper.hpp"
+#include <filesystem>
+#include <iostream>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+/**
+ * @class ZipDecompressor
+ *
+ * @brief Class in charge of decompressing the content in ZIP format as a step of a chain of responsibility.
+ *
+ */
+class ZipDecompressor final : public AbstractHandler<std::shared_ptr<UpdaterContext>>
+{
+private:
+    /**
+     * @brief Pushes the state of the current stage into the data field of the context.
+     *
+     * @param contextData Reference to the context data.
+     * @param status Status to be pushed.
+     */
+    void pushStageStatus(nlohmann::json& contextData, std::string status) const
+    {
+        auto statusObject = nlohmann::json::object();
+        statusObject["stage"] = "ZipDecompressor";
+        statusObject["status"] = std::move(status);
+
+        contextData.at("stageStatus").push_back(std::move(statusObject));
+    }
+
+    /**
+     * @brief Decompress the compressed content and update the context paths.
+     *
+     * @param context Updater context.
+     */
+    void decompress(UpdaterContext& context) const
+    {
+        const auto& outputFolder {context.spUpdaterBaseContext->outputFolder / CONTENTS_FOLDER};
+        std::vector<std::string> newPaths;
+
+        for (const auto& path : context.data.at("paths"))
+        {
+            // Decompress and move paths.
+            auto decompressedFiles {Utils::ZlibHelper::zipDecompress(path, outputFolder)};
+            newPaths.insert(newPaths.end(),
+                            std::make_move_iterator(decompressedFiles.begin()),
+                            std::make_move_iterator(decompressedFiles.end()));
+        }
+
+        // Decompression finished: Update paths.
+        context.data["paths"] = std::move(newPaths);
+    }
+
+public:
+    /**
+     * @brief Decompress the ZIP content and passes the control to the next step on the chain.
+     *
+     * @param context Updater context.
+     * @return std::shared_ptr<UpdaterContext> Next step on the chain.
+     */
+    std::shared_ptr<UpdaterContext> handleRequest(std::shared_ptr<UpdaterContext> context) override
+    {
+        try
+        {
+            decompress(*context);
+        }
+        catch (const std::exception& e)
+        {
+            // Push error state.
+            pushStageStatus(context->data, "fail");
```

Comment:
> Question: Does the status have finite states? (fail and ok)

Yes, only those two values. 

> Don't know if it's used in other places but it could be an enum

It is used all over the content updater components.
A nice solution would be to have some `contentUpdaterHelper.hpp` with the allowed values defined and, for example, the `pushStageStatus` method included.

Reply:
That could be a good option, just to prevent bugs due to typos. But given the state of the module and the ETA it's not the time to address this

Reply:
Thanks for the clarification